src/commands/user-bonuses/list.ts

```diff
@@ -0,0 +1,95 @@
+import * as cosmos from "@azure/cosmos";
+import { Command } from "@oclif/command";
+import cli from "cli-ux";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+
+import { getCosmosConnection, pickAzureConfig } from "../../utils/azure";
+
+export default class UserBonusesList extends Command {
+  public static description = "Lists all user bonuses";
+
+  // tslint:disable-next-line:readonly-array
+  public static args = [
+    {
+      name: "fiscalCode",
+      required: false
+    }
+  ];
+
+  public static flags = {
+    ...cli.table.flags()
+  };
+
+  public async run(): Promise<void> {
+    const { args, flags: parsedFlags } = this.parse(UserBonusesList);
+
+    let fiscalCode: string = "";
+
+    if (args.fiscalCode) {
+      const fiscalCodeOrErrors = FiscalCode.decode(args.fiscalCode);
+      if (fiscalCodeOrErrors.isLeft()) {
+        this.error("Invalid fiscalCode");
+        return;
+      }
+      fiscalCode = fiscalCodeOrErrors.value;
+    }
+
+    try {
+      const config = await pickAzureConfig();
+      cli.action.start("Retrieving cosmosdb credentials");
+      const { endpoint, key } = await getCosmosConnection(
+        config.resourceGroup,
+        config.cosmosName
+      );
+      cli.action.stop();
+
+      cli.action.start("Querying user bonuses...");
+      const client = new cosmos.CosmosClient({ endpoint, auth: { key } });
+      const database = client.database(config.cosmosDatabaseName);
+      const container = database.container(config.cosmosUserBonusesContainer);
+
+      let querySpec = {
+        parameters: [{ name: "@fiscalCode", value: fiscalCode }],
+        query:
+          "SELECT c.fiscalCode, c.isApplicant FROM c WHERE c.fiscalCode = @fiscalCode"
+      };
+
+      if (fiscalCode == "") {
```

Comment:
this should not never happen since the program exits if `fiscalCode` is not in a valid form 

`this.error("Invalid fiscalCode");`

Reply:
ok `return;` is not necessary.
`this.error("Invalid fiscalCode");` is enough to raise the error and exit program

Reply:
```suggestion
      if (fiscalCode == "") {
```
This needs to change query when fiscalCode is not specified